ingress/controllers/nginx/configuration.md

```diff
@@ -0,0 +1,394 @@
+## Contents
+* [Customizing NGINX](#customizing-nginx)
+* [Custom NGINX configuration](#custom-nginx-configuration)
+* [Custom NGINX template](#custom-nginx-template)
+* [Annotations](#annotations)
+* [Custom NGINX upstream checks](#custom-nginx-upstream-checks)
+* [Authentication](#authentication)
+* [Rewrite](#rewrite)
+* [Rate limiting](#rate-limiting)
+* [Secure NGINX upstreams](#secure-nginx-upstreams)
```

Comment:
secure-nginx-upstreams is an annotation name we can't easily adopt to the other controllers. Suggest naming with the assumption that we will bubble these up as fields, no loss if we don't.